ge/org.osate.ge/src/org/osate/ge/internal/services/impl/DefaultActionService.java

```diff
@@ -71,19 +75,23 @@ protected void deactivate() {
 
 	}
 
-	private class ActionGroup implements AgeAction {
+	private class DefaultActionGroup implements AgeAction, ActionGroup {
 		private final String label;
-
+		private final ExecutionMode mode;
+		private final Set<Consumer<Runnable>> undoWrappers; // Consumers which will wrap the execution of the undo actions
 		// Undo actions are the actions needed to undo the original action. Must be in the order
 		// they were originally performed. An action group can be used for redoing as well. In that case, redo is treated as undoing the undo.
 		private final List<AgeAction> undoActions;
 
-		public ActionGroup(final String label) {
-			this(label, new ArrayList<>());
+		public DefaultActionGroup(final String label, final ExecutionMode mode) {
+			this(label, mode, new HashSet<>(), new ArrayList<>());
 		}
 
-		public ActionGroup(final String label, List<AgeAction> actions) {
+		public DefaultActionGroup(final String label, final ExecutionMode mode, final Set<Consumer<Runnable>> wrappers,
+				List<AgeAction> actions) {
 			this.label = Objects.requireNonNull(label, "label must not be null");
+			this.mode = Objects.requireNonNull(mode, "must not be null null");
```

Comment:
Error in message "must not be null null"

Reply:
@RyanMcilnay fixed.